lib/rest-api.php

```diff
@@ -330,3 +330,29 @@ function gutenberg_register_image_editor() {
 	}
 }
 add_filter( 'rest_api_init', 'gutenberg_register_image_editor' );
+
+/**
+ * Registers the post format search handler.
+ *
+ * @param string $search_handlers     Title list of current handlers.
+ *
+ * @return array Title updated list of handlers.
+ */
+function gutenberg_post_format_search_handler( $search_handlers ) {
+	$search_handlers[] = new WP_REST_Post_Format_Search_Handler();
```

Comment:
I think we should probably only register this if `current_theme_supports( 'post-formats' )`.

Reply:
This means that calling `GET /v2/search` with `type=post-format` will result in a 400 error unless the client checks `OPTIONS /v2/search` first. That's a little cumbersome, no?

Reply:
I'm not sure. I think it's probably worth doing an `OPTIONS` request always to be able query other sources that the client doesn't already know about.

But it seems like the preflight check could be skipped since a `400` response with an error for that parameter could be treated as an empty result sense and potentially excluded from future queries.

Reply:
In d44950bbd3 I made it so that the handler isn't registered and Gutenberg won't call `/v2/search?type=post-format` if the current theme doesn't support post formats.